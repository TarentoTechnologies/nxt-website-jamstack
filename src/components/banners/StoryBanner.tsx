import { GatsbyImage, getImage } from "gatsby-plugin-image";
import ReactHtmlParser from "react-html-parser";

import { carbon } from "../../styles/style-guide/ColorGuide.module.css";
import { PrimaryCTA } from "../buttons/PrimaryCTA";
import { PrimaryTitle } from "../titles/PrimaryTitle";
import {
  bannerStyles,
  buttonStyles,
  customH3,
  imgStyles,
  innovationStory,
  subFont,
  subTitleStyles,
  techSolution,
  titleStyles,
  whiteBox,
} from "./StoryBanner.module.css";

interface StoryBannerProps {
  primaryTitle: string;
  cardData: any;
  subTitle?: string;
  bgImg?: any;
  variant?: string;
  withCTA?: boolean;
  isImage?: boolean;
  spaceAboveCTA?: boolean;
}

export const StoryBanner = ({
  primaryTitle,
  cardData,
  variant,
  bgImg,
  subTitle,
  withCTA,
  isImage,
  spaceAboveCTA,
}: StoryBannerProps) => {
  const image: any = getImage(cardData?.Image?.localFile);

  return (
    <div
      className={`container-fluid ${bannerStyles} d-flex justify-content-center`}
      style={
        variant === "lead"
          ? {
              backgroundImage: `url(${bgImg})`,
              backgroundRepeat: "no-repeat",
              backgroundSize: "cover",
            }
          : {}
      }
      id={cardData?.id}
    >
      <div className={`row containerService paddingLeftRight15`}>
        <div className={`col-12 ${titleStyles} px-lg-0`}>
          <PrimaryTitle title={primaryTitle} variant="h1700" color="white" />
          <h3 className={`${subTitleStyles}`} data-aos="fade-up">
            {subTitle}
          </h3>
        </div>
        <div
          className={`col-lg-6 ${imgStyles} d-flex justify-content-center align-items-center`}
        >
          {!isImage ? (
            <img
              src={cardData?.Image?.localFile?.url}
              alt={
                cardData?.Image?.alternativeText
                  ? cardData?.Image?.alternativeText
                  : ""
              }
              title={cardData?.Image?.caption ? cardData?.Image?.caption : ""}
              className={`img-fluid`}
              data-aos="fade-up"
            />
          ) : (
            <GatsbyImage
              image={image}
              alt={
                cardData?.Image?.alternativeText
                  ? cardData?.Image?.alternativeText
                  : ""
              }
              className={`img-fluid`}
              data-aos="fade-up"
            />
          )}
        </div>
        <div
          className={`col-lg-6 d-flex justify-content-center align-items-center`}
        >
          <div
            className={`${whiteBox} ${carbon} p-5
            ${variant === "tech" ? techSolution : ""}
            ${variant === "innovation" ? innovationStory : ""}`}
          >
            {cardData?.PrimaryTitle && (
              <h3
                className={`${customH3} customMaringBtm1 pt-4 pb-3`}
                data-aos="fade-up"
              >
                {cardData?.PrimaryTitle}
              </h3>
            )}
            <div
              className={`${spaceAboveCTA ? "pb-4" : ""}`}
              data-aos="fade-up"
            >
              {ReactHtmlParser(
                cardData?.Description.data.childMarkdownRemark.html
              )}
            </div>
            {withCTA && (
              <div className={`${variant === "lead" ? buttonStyles : ""}`}>
                <PrimaryCTA
                  label={cardData?.CTAText}
                  btnLink={cardData?.CTALink}
                  isMail={cardData?.isCTAEmail}
                  size="large"
                />
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
