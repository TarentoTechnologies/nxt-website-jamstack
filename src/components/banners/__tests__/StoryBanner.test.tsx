import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { StoryBanner } from "../StoryBanner";

const data = {
  SolutionTitle: "Tech Solutions",
  SolutionCard: {
    CTALink: "hello@tarento.com",
    CTAText: "Find out more",
    Description: {
      data: {
        childMarkdownRemark: {
          html: "<p><strong>Empowering customer experience with data &#x26; Insights</strong></p>\n<p>Aurora customer experience platform brings in the best of data engineering, analytics and user experience design to enable a superior customer support experience.</p>",
        },
      },
    },
    Image: {
      id: "68965cf5-4362-522b-88cb-cfe574af79f2",
      localFile: {
        childImageSharp: {
          gatsbyImageData: {
            height: 858,
            images: {
              fallback: {
                sizes: "(min-width: 934px) 934px, 100vw",
                src: "/static/4af750352a6c177c7f80d9ed351d244d/6998c/tech_solution_d8d031719f.png",
                srcSet:
                  "/static/4af750352a6c177c7f80d9ed351d244d/fd487/tech_solution_d8d031719f.png 234w,\n/static/4af750352a6c177c7f80d9ed351d244d/a2a8c/tech_solution_d8d031719f.png 467w,\n/static/4af750352a6c177c7f80d9ed351d244d/6998c/tech_solution_d8d031719f.png 934w",
              },
              layout: "constrained",
              placeholder: {
                fallback:
                  "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAASCAYAAABb0P4QAAAACXBIWXMAAAsTAAALEwEAmpwYAAAEZUlEQVR42oWUe0xTVxzHK8ZZE83UMRhKtCD0cdvbJ31Q2t4WsVAy3EMv1g7C4qPogFIeRdSx+gjQFstDh5MQh3HL5irLCPFBVKpsC9syt0SXZTHZsmXJ3B9bSEyWbJH2fndui8SMP3aSb36/e3LP53y/J/dcgWBhsAJ2OV/N8ojTogjft9DhKasy/K5FGQpa6cjrJao+u4XqpsyS8AaTKbqKf9c7ghVtZ5HVEPhLu6fpN+eehodFgkUgG0sBGXpgt0NzBox6MGnXDMKuHiA9r35Y6Mg82eh3izJy36oI3WLUp8bLmYsz21+87ovFsFzw9GDZtEMbHXJZVWFYlT2PzfKe+RJeit55izKlpFUVgk3VBxvZwKaKJvjNzPLeoZRj790Vi0D+YWwMwhLloMWh+BAOVQxV2yfhckxgq+YyHHQMDkWMc9AfJ+2K0UQp1ZtwUn2Py2VRbBdHv6vOj7pToKAgmMHXV5gxU6khVG9RDCutsnHY5ddRYbuGSsc1lBVNYlvRFTI3AUY+CfuWMZRmHYcju4fbI/NzL+X1/b0152xfGhhMA9mykWdJyWCMX4jKVee5KvUIWfwR3OpemMUXYFddAkN/ACt1CfrcM1CsPARbVivHPOeHWtg1pxD2tC7GhQDL+Nq4+7a0tupGWyl1+p8KahBm2XucYcsIigtHYcg/R/QOisWjMOWeArXiMFyiZo5eFYBuTWiuXHTRl4JVVl7Z3N32o+J48wNjV+M9V43r6g5T4YVHduU4rLJRTrspCn3+aehE/dDnDaNE8j50L/SjLMeLHQU1nG71URRnDv9hzjy3KwV0Oi+bQ20P3CdbfzgSDvz02jblxSy9aOhX7aYI6I0nOFVuN5Qbu0l/EkWiQVill6DNjsC4tgOOnBbo1xxOuvImHro1sxWLkZ+co5e9afG+OnWCyg78Is5shyy7g5NktYPeECBKPYPK6YB8XQckgkPQP/8mVKvbEzLBkT9VK0860+cHwbJgEBksm/4w9748ZSjTvT1nEndDK+riDAVDMOSFSexuAjyKgswmyAhQu5bM5fZCuT6QVGd3QlfQNsswdcIlN6XRM1vmq/0KB3bFuX07bmD/zhl4XNfhdl5FhWEMxZIIdPnHyIc/AL30MHQSX9JId5C+/XtTbvpKLsRGKvZBNs7Us3ECugUvO010i4cTTeOg+w6ZvwmLqguawgYUSVtQJPEnDdIA6f33mc1B4RIgf44Hds2gvjqeAj6pT8S7tqrfglbcTCCt0ElbEgYqDSwoaFq5FOiOG3kIUcLLxpML/aL2E8dpoC8NlPjnDbIA9LK2exQVfGYJ0Of5Uuev/RbNNXfxxu7PSdTbSIOn58lRJBeAnFbclCBAzkB1JooVXXz8T8nyZU/9b9K35aDnyjoSrYss/oTUn73V8WSDZxa+mq/BV96lTXOMOOJdtSZJ1M90spa9Jiq4XvB/o67uttDruUPXs9P7yQYjRN/s23njMXH4SF3YeN4o6zT+x5XgX7vwMW0BNcjuAAAAAElFTkSuQmCC",
              },
              width: 934,
            },
          },
        },
      },
    },
  },
  withCTA: true,
  variant: "tech",
  isImage: true,
};

describe("Test if the StoryBanner component is working as expected", () => {
  it("<StoryBanner /> matches snapshot", () => {
    const component = render(
      <StoryBanner
        primaryTitle={data.SolutionTitle}
        cardData={data.SolutionCard}
        withCTA={data.withCTA}
        variant={data.variant}
        isImage={data.isImage}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<StoryBanner /> renders without exploding", () => {
    const component = render(
      <StoryBanner
        primaryTitle={data.SolutionTitle}
        cardData={data.SolutionCard}
        withCTA={data.withCTA}
        variant={data.variant}
        isImage={data.isImage}
      />
    );
    expect(component.getByTestId("story-banner")).toBeInTheDocument();
  });
});
