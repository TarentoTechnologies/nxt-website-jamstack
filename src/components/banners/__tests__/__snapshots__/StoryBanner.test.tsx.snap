// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the StoryBanner component is working as expected <StoryBanner /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid bannerStyles d-flex justify-content-center"
    data-testid="story-banner"
  >
    <div
      class="row containerService paddingLeftRight15"
    >
      <div
        class="col-12 titleStyles px-lg-0"
      >
        <div
          class="
      undefined
        
        
        
        
        
        
        
        h1700
        
        
        white"
          data-aos="fade-up"
          data-testid="primary-title"
        >
          Tech Solutions
        </div>
        <h3
          class="subTitleStyles"
          data-aos="fade-up"
        />
      </div>
      <div
        class="col-lg-6 imgStyles d-flex justify-content-center align-items-center"
      >
        <div
          class="gatsby-image-wrapper img-fluid"
          data-gatsby-image-wrapper=""
          style="position: relative; overflow: hidden;"
        >
          <div
            aria-hidden="true"
            data-placeholder-image=""
            style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          />
          <img
            alt=""
            data-aos="fade-up"
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="/static/4af750352a6c177c7f80d9ed351d244d/6998c/tech_solution_d8d031719f.png"
            data-srcset="/static/4af750352a6c177c7f80d9ed351d244d/fd487/tech_solution_d8d031719f.png 234w,/static/4af750352a6c177c7f80d9ed351d244d/a2a8c/tech_solution_d8d031719f.png 467w,/static/4af750352a6c177c7f80d9ed351d244d/6998c/tech_solution_d8d031719f.png 934w"
            decoding="async"
            loading="lazy"
            sizes="(min-width: 934px) 934px, 100vw"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
          <noscript />
          <script
            type="module"
          >
            const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
          </script>
        </div>
      </div>
      <div
        class="col-lg-6 d-flex justify-content-center align-items-center"
      >
        <div
          class="whiteBox carbon p-5
            techSolution
            "
        >
          <div
            class=""
            data-aos="fade-up"
          >
            <p>
              <strong>
                Empowering customer experience with data & Insights
              </strong>
            </p>
            <p>
              Aurora customer experience platform brings in the best of data engineering, analytics and user experience design to enable a superior customer support experience.
            </p>
          </div>
          <div
            class=""
          >
            <a
              data-testid="primary-cta"
              href="hello@tarento.com"
            >
              <div
                class="text-center d-flex align-items-center justify-content-center 
      secondaryBtn large undefined"
                data-aos="fade-up"
              >
                <label
                  class="labelStyles"
                >
                  Find out more
                </label>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
